libs/MailMessage/src/MailMessageBuilder.cpp

```diff
@@ -0,0 +1,68 @@
+#include "MailMessageBuilder.h"
+
+#include "MailAddress.h"
+#include "MailMessage.h"
+#include "MailAttachment.h"
+
+namespace ISXMM
+{
+MailMessageBuilder& MailMessageBuilder::set_from(const std::string& email, const std::string& name)
+{
+    m_from = { email, name };
+    return *this;
+}
+
+MailMessageBuilder& MailMessageBuilder::add_to(const std::string& email, const std::string& name)
+{
+    m_to.push_back({ email, name });
+    return *this;
+}
+
+MailMessageBuilder& MailMessageBuilder::add_cc(const std::string& email, const std::string& name)
+{
+    m_cc.push_back({ email, name });
+    return *this;
+}
+
+MailMessageBuilder& MailMessageBuilder::add_bcc(const std::string& email, const std::string& name)
+{
+    m_bcc.push_back({ email, name });
+    return *this;
+}
+
+MailMessageBuilder& MailMessageBuilder::set_subject(const std::string& subject)
+{
+    m_subject = subject;
+    return *this;
+}
+
+MailMessageBuilder& MailMessageBuilder::set_body(const std::string& body)
+{
+    m_body = body;
+    return *this;
+}
+
+MailMessageBuilder& MailMessageBuilder::add_attachment(const std::string &path)
+{
+    try
+    {
+        m_attachments.push_back({ path }); 
+    }
+    catch(const std::exception& e)
+    {
+        throw e;
```

Comment:
Why rethrowing the exception instead of handling it here? In main() you are wrapping that in try-catch but in the test you don't and you don't use any EXPECT_THROW. Is this a critical error? You could handle it printing that adding attachement failed and proceed with MailMessage without so this method user won't have to handle the exception and end user might try to re-attach the file.